skfda/preprocessing/dim_reduction/feature_extraction/_per_class_transformer.py

```diff
@@ -178,10 +235,16 @@ def transform(self, X: T) -> Union[DataFrame, np.ndarray]:
         """
         sklearn_check_is_fitted(self)
 
-        transformed_data = np.empty((len(X), 0))
-        for feature_transformer in self._class_feature_transformers_:
+        for j, feature_transformer in enumerate(
+            self._class_feature_transformers_,
+        ):
             elem = feature_transformer.transform(X)
             data = np.array(elem)
```

Comment:
Why? Why not simply putting every column in a list using a list comprehension and THEN converting to array/dataframe?

Reply:
I have just changed it to a list comprehension.
While creating a ndarray as output with np.array(transformed_data) an exception is risen: ValueError: could not broadcast input array from shape (x) into shape (y), and so I have also changed the way the ndarray is created.